### PR TITLE
Validate Collection Folders on adding and removal

### LINKED ISSUE #11258: [Issue]: 8 Empty Collections "Collections".."Collections8"
### Please describe your bug

I just set up Jellyfin and it keeps creating 8 collections:

image

I can delete these but then they Re-appear. Auto create collections is off for both my Movies and Music libraries.

### Reproduction Steps

Go to dashboard

### Jellyfin Version

Unstable (master branch)

### if other:

_No response_

### Environment

```markdown
- OS: Ubuntu 22
- Linux Kernel: aarch64 Linux 5.10.160-legacy-rk35xx
- Virtualization: Docker
- Clients: Browser (probably all the others)
- Browser: Chrome 120
- FFmpeg Version: 6.0.1-1+git20240229.99d938d6~jammy
- Playback Method: N/A
- Hardware Acceleration: N/A
- GPU Model: Rockchip GPU on Orange PI 5B
- Plugins: Default plugins
- Reverse Proxy: Cloudflare tunnels
- Base URL: https://jf.[MY_DOMAIN]/
- Networking: ??
- Storage: lots
```

### Jellyfin logs

Sometimes I get error refreshing item but nothing collections related it seems:
```shell
[2024-03-29 02:02:59.793 +00:00] [ERR] [50] MediaBrowser.Providers.Manager.ProviderManager: Error refreshing item
System.InvalidOperationException: Cannot update refresh progress of item 'MusicArtist' (5e6effafcbd46ccf500c0bf0d78b5965) because a refresh for this item is not running
   at MediaBrowser.Providers.Manager.ProviderManager.<>c__DisplayClass72_0.<OnRefreshProgress>b__0(Guid _)
   at System.Collections.Concurrent.ConcurrentDictionary`2.AddOrUpdate(TKey key, Func`2 addValueFactory, Func`3 updateValueFactory)
   at MediaBrowser.Providers.Manager.ProviderManager.OnRefreshProgress(BaseItem item, Double progress)
   at MediaBrowser.Controller.Entities.Audio.MusicAlbum.RefreshAllMetadata(MetadataRefreshOptions refreshOptions, IProgress`1 progress, CancellationToken cancellationToken)
   at MediaBrowser.Controller.Entities.Folder.<>c__DisplayClass67_0.<<RefreshAllMetadataForContainer>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at MediaBrowser.Providers.Manager.ProviderManager.RunMetadataRefresh(Func`1 action, CancellationToken cancellationToken)
   at MediaBrowser.Controller.Entities.Folder.RefreshAllMetadataForContainer(IMetadataContainer container, MetadataRefreshOptions refreshOptions, IProgress`1 progress, CancellationToken cancellationToken)
   at MediaBrowser.Controller.Entities.Folder.RefreshChildMetadata(BaseItem child, MetadataRefreshOptions refreshOptions, Boolean recursive, IProgress`1 progress, CancellationToken cancellationToken)
   at MediaBrowser.Controller.Entities.Folder.<>c__DisplayClass70_0`1.<<RunTasks>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at MediaBrowser.Controller.Entities.Folder.RunTasks[T](Func`3 task, IList`1 children, IProgress`1 progress, CancellationToken cancellationToken)
   at MediaBrowser.Controller.Entities.Folder.ValidateChildrenInternal2(IProgress`1 progress, Boolean recursive, Boolean refreshChildMetadata, MetadataRefreshOptions refreshOptions, IDirectoryService directoryService, CancellationToken cancellationToken)
   at MediaBrowser.Controller.Entities.Folder.ValidateChildrenInternal(IProgress`1 progress, Boolean recursive, Boolean refreshChildMetadata, MetadataRefreshOptions refreshOptions, IDirectoryService directoryService, CancellationToken cancellationToken)
   at MediaBrowser.Providers.Manager.ProviderManager.RefreshArtist(MusicArtist item, MetadataRefreshOptions options, CancellationToken cancellationToken)
   at MediaBrowser.Providers.Manager.ProviderManager.StartProcessingRefreshQueue()
```

### FFmpeg logs

N/A

### Please attach any browser or client logs here

N/A

### Please attach any screenshots here

See above

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #11385: [bug]:  Collection is not created
### Please describe your bug

When trying to create a collection a folder "collection" is created and the GUI is showing spinning wheel, but freezes. Restarting the app and trying to  create a new collection creates a new folder "collection2", same result in GUI. 

https://forum.jellyfin.org/t-10-9-0-collection-are-not-created?pid=20906#pid20906

### Reproduction Steps

1. extract jellyfin_2024041505-amd64.zip to Programfolder
2. run with new datadir "C:\Program Files\Jellyfin\Server.10.9\jellyfin.exe" --service --datadir "D:\JellyfinData"
3. add movie library
4. restart
5. add collection via Movies / Collections / New Collection

### Jellyfin Version

10.9.0

### if other:

jellyfin_2024041505-amd64.zip

### Environment

```markdown
- OS: Windows 11 Pro, 22H2, build 22621.3447
- Virtualization: no
- Clients: Jellyfin Media Player 1.9.1
- FFmpeg Version: default
- Playback Method: no playback
- Hardware Acceleration: no playback
- GPU Model: Iris Xe Graphics (Intel Core I7-1165G7 11th Gen.)
- Plugins: none
- Reverse Proxy: Apache 2.4.58
- Base URL: http://jellyfin.domain.tld
- Networking: local
- Storage: NAS/rclone with local link
```

### Jellyfin logs

```shell
for log see post on forum
https://my.hidrive.com/lnk/OZBhCibCU

Posting here gives: 

There was an error creating your issue: body is too long, body is too long (maximum is 65536 characters).
```

### FFmpeg logs

_No response_

### Please attach any browser or client logs here

_No response_

### Please attach any screenshots here

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by gnattu**
**Changes**

This PR validates the collection folder on addition and removal. This folder is very special and is hard to follow the common virtual folder validation. It needs to skip the accessible validation on addition to be successfully added, and it will require the accessible validation to actually remove this virtual folder. The current approach is to use the `allowRemoveRoot` parameter to determine if the path accessible check is needed, because when `allowRemoveRoot == true`, it means the user explicitly requires a deletion.

**Issues**

Fixes #11258
Fixes #11385

### CODE REVIEW

**Review by joshuaboniface — APPROVED** [review]

**Review by crobibero — APPROVED** [review]

File: MediaBrowser.Controller/Entities/Folder.cs [hunk 1]
```diff
@@ -333,8 +333,13 @@ protected virtual async Task ValidateChildrenInternal(IProgress<double> progress
             }
         }
 
-        private static bool IsLibraryFolderAccessible(IDirectoryService directoryService, BaseItem item)
+        private static bool IsLibraryFolderAccessible(IDirectoryService directoryService, BaseItem item, bool checkCollection)
         {
+            if (!checkCollection && (item is BoxSet || string.Equals(item.FileNameWithoutExtension, "collections", StringComparison.OrdinalIgnoreCase)))
```
Comment (nielsvanvelzen) [inline MediaBrowser.Controller/Entities/Folder.cs:338]: The folder name may be localized, see

https://github.com/jellyfin/jellyfin/blob/169698619dfe2989d0562b9723d0a01cfa02b9ba/Emby.Server.Implementations/Collections/CollectionManager.cs#L103

**Review by nielsvanvelzen — APPROVED** [review]
